### PR TITLE
Make working copy get diff for update from `MergedTree`

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: lib/src/merged_tree.rs [hunk 1]
```diff
@@ -504,3 +518,200 @@ impl Iterator for ConflictIterator {
...
+        MergedTreeValue<'a>,
+        MergedTreeValue<'a>,
+    );
+
+    fn next(&mut self) -> Option<Self::Item> {
+        for basename in self.basename_iter.by_ref() {
+            let value_before = self.before.value(basename);
+            let value_after = self.after.value(basename);
+            if value_after != value_before {
+                return Some((basename, value_before, self.after.value(basename)));
```
Comment (yuja) [inline lib/src/merged_tree.rs:574]: nit: `value_after` instead of `self.after.value...`

File: lib/src/working_copy.rs [hunk 2]
```diff
@@ -1228,6 +1232,9 @@ impl TreeState {
          -> Result<(), CheckoutError> {
             let disk_path = path.to_fs_path(&self.working_copy_path);
 
+            if before.is_present() {
+                fs::remove_file(&disk_path).ok();
+            }
```
Comment (yuja) [inline lib/src/working_copy.rs:1237]: redundant `fs::remove_file()` in `Ok(None)` case

**Review by yuja — APPROVED** [review]
Looks good, but I don't have expertise on tree-level conflicts. @ilyagr, can you take a quick look?

File: lib/src/merge.rs [hunk 3]
```diff
@@ -415,6 +415,13 @@ impl Merge<Option<TreeValue>> {
         backend::Conflict { removes, adds }
     }
 
+    pub fn is_tree(&self) -> bool {
```
[the remark this answers is not in the document]
Comment (martinvonz) [reply lib/src/merge.rs:419]: Yes, this definitely should have a comment. I've added one along the lines of what you said. I'll try to remember to send a separate PR removing the `#![allow(missing_docs)]` too.

File: lib/src/merged_tree.rs (hunk 1 above)
Comment (martinvonz) [reply lib/src/merged_tree.rs]: Thanks for catching that.

File: lib/src/working_copy.rs (hunk 2 above)
Comment (martinvonz) [reply lib/src/working_copy.rs:1237]: Fixed, thanks.

**Comment by ilyagr** [comment]
I got in touch with Martin over chat to warn him that I'm slow to review at the moment; sorry I didn't get to this in time.

**Pull request merged**